src/plugins/intel_gpu/src/plugin/transformations/convert_matmul_to_fc.cpp

```diff
@@ -41,10 +41,21 @@ ConvertMatMulToFullyConnected::ConvertMatMulToFullyConnected() {
             return false;
         }
 
-        // fc_input_a and fc_input_b - are the final inputs that will be set to FullyConnected.
-        // So in case of adding new operations that takes matmul inputs we need keep update fc_input_a and fc_input_b.
         auto fc_input_a = pattern_map.at(activations_m);
         auto fc_input_b = pattern_map.at(weights_m);
+
+        // Not to transpose fc_input_b which should be fc_input_a of other sibling MatMul
+        auto input_b = fc_input_b.get_node_shared_ptr();
+        for (auto& user : input_b->get_users()) {
+            if (user != matmul && ov::is_type<ov::op::v0::MatMul>(user)) {
+                auto other_matmul = std::dynamic_pointer_cast<ov::op::v0::MatMul>(user);
+                if (input_b == other_matmul->get_input_node_shared_ptr(0))
+                    return false;
+            }
+        }
+
+        // fc_input_a and fc_input_b - are the final inputs that will be set to FullyConnected.
+        // So in case of adding new operations that takes matmul inputs we need keep update fc_input_a and fc_input_b.
```

Comment:
Could you please implement a test case?

I think we need to have more test to understand the impact of this change.
For example, input is connected to input1 of two matmuls(A, B) and transpose configuration may be different from A and B.